test/constants.js

```diff
@@ -26,12 +26,24 @@ const jwks = {
         '7KRDtHuJ9-R1cYzB9-E4TUVazzv93MMmMo_38nOwEKNxlWs7OVg397d0SCsdmBbcbr4KTMeblY4a-VOzLVZ5ycYgi7ZbMvv7RzunKuPsjm7m863dLnPUFOptsFVANDOHgDYopKBFYoIMoxjXU7bOzLL-Ez0oO5keT1hGZkJT_7GRvKyYigugN4lLia4Tb3AmUN60wiloyQCJ2xYATWHB0e4sTwIDq6MFXhVFHXV6ZBU7sDh0HqmP08gJtMnsFOE7zUcbpqTvpz5nAR6EyUs7R0g61WmGUfQTrE6byVCZ8w0NN4Xer6IQBjnDZWbmf69jsAFFAYDCe-omWXY526qLQw',
       use: 'sig',
     },
+    {
+      kid: 'KEY_2',
+      e: 'AQAB',
+      kty: 'RSA',
+      n:
+        '7KRDtHuJ9-R1cYzB9-E4TUVazzv93MMmMo_38nOwEKNxlWs7OVg397d0SCsdmBbcbr4KTMeblY4a-VOzLVZ5ycYgi7ZbMvv7RzunKuPsjm7m863dLnPUFOptsFVANDOHgDYopKBFYoIMoxjXU7bOzLL-Ez0oO5keT1hGZkJT_7GRvKyYigugN4lLia4Tb3AmUN60wiloyQCJ2xYATWHB0e4sTwIDq6MFXhVFHXV6ZBU7sDh0HqmP08gJtMnsFOE7zUcbpqTvpz5nAR6EyUs7R0g61WmGUfQTrE6byVCZ8w0NN4Xer6IQBjnDZWbmf69jsAFFAYDCe-omWXY526qLQw',
+      use: 'sig'
+    }
   ],
 }
 const privateKey = readFileSync(path.join(__dirname, 'private.pem'), 'utf8')
 const jwk = jwks.keys[1]
+let defaultSigningAlgorithm;
+if (jwk.alg === undefined && jwk.kty === 'RSA') {
+  defaultSigningAlgorithm = 'RS256';
+}
 const token = jwt.sign({ name: 'Jane Doe' }, privateKey, {
-  algorithm: jwk.alg,
+  algorithm: jwk.alg || defaultSigningAlgorithm,
```

Comment:
as far as I understand this code is doing nothing. am I missing something? can you try to remove it and see what happens?

Reply:
If you replace `keys[1]` by `keys[2]`, then jwt.sign will fail because there is no algorithm to sign with (_Error: "algorithm" must be a valid string enum value_) ; I had the idea that we could perform the test with differents keys, but since I was mistaken I've removed it 🙂